### PR TITLE
refactored after review

### PR DESCRIPTION
**Pull request by elinakolod**

### CODE REVIEW

File: .rake_tasks~ [hunk 1]
```diff
@@ -0,0 +1,52 @@
+about
```
Comment (yuriy1991) [inline .rake_tasks~]: please add this file to `.gitignore` and remove from git

File: app/admin/dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  content title: proc { I18n.t('active_admin.dashboard') } do
+    div class: 'blank_slate_container', id: 'dashboard_default_message' do
+      span class: 'blank_slate' do
+        span I18n.t('active_admin.dashboard_welcome.welcome')
+        small I18n.t('active_admin.dashboard_welcome.call_to_action')
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (yuriy1991) [inline app/admin/dashboard.rb]: please remove comments

File: app/assets/javascripts/cable.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Action Cable provides the framework to deal with WebSockets in Rails.
```
Comment (yuriy1991) [inline app/assets/javascripts/cable.js]: are you using web sockets? if no, please delete this file and an empty channels folder

File: app/assets/javascripts/comments.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          $(obj).addClass(checked);
+        }else{
+          $(obj).addClass(empty);
+        }
+      })
+      let rating = $('.rating').children('.rate-star').length;
+      $('#comment_form_rating').val(rating);
+    });
+  }
+});
```
Comment (yuriy1991) [inline app/assets/javascripts/comments.js]: no new line at the end of file

File: app/assets/stylesheets/authors.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Authors controller here.
```
Comment (yuriy1991) [inline app/assets/stylesheets/authors.scss]: please remove this file

File: app/assets/stylesheets/books.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Books controller here.
```
Comment (yuriy1991) [inline app/assets/stylesheets/books.scss]: please remove this file

File: app/assets/stylesheets/carts.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Carts controller here.
```
Comment (yuriy1991) [inline app/assets/stylesheets/carts.scss]: please remove all empty scss files)

File: app/controllers/addresses_controller.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class AddressesController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :authenticate_user!
+
+  respond_to :js
+
+  def create
```
Comment (yuriy1991) [inline app/controllers/addresses_controller.rb]: It looks like we are updating the address here. Maybe it should be called `def update`?

File: app/controllers/users/passwords_controller.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class Users::PasswordsController < Devise::PasswordsController
+  def create
```
Comment (yuriy1991) [inline app/controllers/users/passwords_controller.rb]: Devise gem allows to customize it's behaviour by passing a block, that will be yielded.
Can we remove the copy-pasted devise methods and change the behavior like this?
```
def create
    super do |resource|
      # your custom logic
    end
  end
```

File: app/forms/address_form.rb [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+  private
+
+  def update
+    @user = User.find(addressable_id)
+    update_users_address(type)
+  end
+
+  def update_users_address(type)
+    @user.send("build_#{type}_address".to_sym) unless @user.send("#{type}_address".to_sym)
```
Comment (yuriy1991) [inline app/forms/address_form.rb]: can we use `public_send` here? it's dangerous to use `send`

File: app/services/popular_books_service.rb [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+  def call
+    group_by_category(@books)
+    decorate_books
+  end
+
+  def group_by_order_items_quantity(books)
+    grouped_books = {}
+    books.each { |book| grouped_books[book] = book.order_items.sum(&:quantity) }
+    grouped_books.sort_by { |_book, order_items_number| order_items_number }.reverse.to_h
```
Comment (yuriy1991) [inline app/services/popular_books_service.rb]: I think reverse is not needed
```
grouped_books.sort_by { |_book, order_items_number| -order_items_number }.to_h
```

File: app/views/admin/categories/edit.js.erb [hunk 12]
```diff
@@ -0,0 +1,4 @@
+if ( $( '.formtastic.category' ).length ){
+    $( '.formtastic.category').remove();
+}
+$('#main_content').prepend("<%= j render partial: 'form', locals: {category: @category} %>");
```
Comment (yuriy1991) [inline app/views/admin/categories/edit.js.erb]: no new line

File: app/views/admin_users/registrations/edit.html.erb [hunk 13]
```diff
@@ -0,0 +1,43 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
```
Comment (yuriy1991) [inline app/views/admin_users/registrations/edit.html.erb]: I18n

File: config/database.yml [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  
+
+development:
+  <<: *default
+  database: bookstore
+  username: <%= Rails.application.credentials.dig(:db, :username) %>
+  password: <%= Rails.application.credentials.dig(:db, :password) %>
+  host: localhost
+  # The specified database role being used to connect to postgres.
```
Comment (yuriy1991) [inline config/database.yml]: Please remove the unnecessary comments

File: app/services/admin/admin_utils_service.rb [hunk 15]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class Admin::AdminUtilsService
```
Comment (yuriy1991) [inline app/services/admin/admin_utils_service.rb]: This doesn't look like a service object. Maybe we should move this to helpers?

File: app/services/cart_utils_service.rb [hunk 16]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CartUtilsService
+  class << self
+    def subtotal(order_items)
```
Comment (yuriy1991) [inline app/services/cart_utils_service.rb]: These methods should be removed from services. `Order` model?

**Comment by yuriy1991** [comment]
Please check the styles and layout. It doesn't match the mockups:
http://joxi.ru/823pLJKS9Jvq7A
Also, I wasn't able to sign up using the Facebook

File: app/controllers/users/passwords_controller.rb (hunk 9 above)
Comment (elinakolod) [reply app/controllers/users/passwords_controller.rb]: @yuriy1991 Thank you! You've helped me a lot!

File: app/services/admin/admin_utils_service.rb (hunk 15 above)
Comment (elinakolod) [reply app/services/admin/admin_utils_service.rb]: @yuriy1991 moved this methods to ActiveAdminHelper

File: .rake_tasks~ (hunk 1 above)
Comment (elinakolod) [reply .rake_tasks~]: @yuriy1991 all done

**Pull request merged**